static/app/components/charts/worldMapChart.tsx

```diff
@@ -46,136 +37,118 @@ const DEFAULT_CENTER_X = 10.97;
 const DISCOVER_QUERY_LIST_CENTER_Y = -12;
 const DEFAULT_CENTER_Y = 9.71;
 
-class WorldMapChart extends React.Component<Props, State> {
-  state: State = {
-    countryToCodeMap: null,
-    map: null,
-    codeToCountryMap: null,
-  };
+export function WorldMapChart({
+  series,
+  seriesOptions,
+  fromDiscover,
+  fromDiscoverQueryList,
+  ...props
+}: WorldMapChartProps) {
+  const map = worldMap;
+  const codeToCountryMap = Object.fromEntries(
```

Comment:
Iirc we had some crashes when using fromEntries. I would rewrite this into smth like

```ts
const codeToCountryMap = Record<string, string>
for (const country in map) {
 codeToCountryMap[countryToCodeMap[country]] = country
}
```

Reply:
sure..let me update it...thanks!